hack/generate-proto.sh

```diff
@@ -10,6 +10,8 @@ header "generating proto files"
 ensure_vendor
 
 make_fake_paths
+# This overrides the existing trap
+trap 'cd "${FAKE_GOPATH}" && go clean -modcache && cd /tmp && rm -rf "${FAKE_GOPATH}"' EXIT
```

Comment:
@tmenjo - The permission issue only happens when proto-gen fails right? So this change only cleans up the files, but does not solve your proto-gen failure, is my understanding correct?

Reply:
In my words, the permission issue caused ./hack/generate-proto.sh to exit with non-zero code. Then it led to the failure of  `make codegen` and `make pre-push -B`.

When I removed (for testing purpose only) the existing `rm -rf ${FAKE_GOPATH}` trap registration in ./hack/library.sh, then ./hack/generate-proto.sh exited with 0. It meant that protoc or other commands in the script succeeded, though there are some warnings.

So the simple `rm -rf ${FAKE_GOPATH}` looked the root cause to me.

Reply:
hmmmm... I kind of know what was happening... I'll merge your PR, but will have a followup fix/dependency upgrade , and will ping you to see if that works in your env.